### PR TITLE
Seismic interpolation cleanup

### PR DESCRIPTION
**Pull request by mlange05**
So it turns out that all the explicit time-index replacements via the outdated `u_t` and `p_t` arguments are pretty unnecessary. This merge removes them and uses `field=u.forward / v.backward` instead, as well as defaulting to the `time` dimension in the `PointData` constructor.

### CODE REVIEW

**Comment by FabioLuporini** [comment]
OK

**Comment by mlange05** [comment]
I can't actually see any use case that you can't use with `u.forward/backward`, except for when you sample/interpolate the src/rec data at a time offset. For this very specific case you can still use a simple `subs` in the operator definition code. In the meantime, I think it's much more dangerous to have outdated and unused backdoors lying around that clutter up the interface.

**Comment by mloubout** [comment]
>I can't actually see any use case that you can't use with u.forward/backward

Current 4th order, `u.forward` is `t+2` which is not the one you want (`t, t+1, t-1`  only in propagation once #298 is fixed). I actually have no idea why TRAVI is green as 4th order in time should fail with this implementation.

This also gonna be problematic for elastic (first order coupled) where either `forward` or `backward` is gonna create problem.

**Comment by mlange05** [comment]
Is this not a separate problem? Regardless of fixing the 4th order `u`/`v` discretization, should the time index we update with the stencil not be the same as the time index used in src/rec expressions? If so, I would say this is an unrelated issue.

> This also gonna be problematic for elastic (first order coupled) where either forward or backward is gonna create problem.

You need to be way more specific than that. In the meantime you can manually sub indices on the input to `interpolate` and the discretized output epxpression, so unless we can specify what the problem is there should be enough flexibility in the interface to create whatever you need.

**Comment by mloubout** [comment]
>should the time index we update with the stencil not be the same as the time index used in src/rec expressions? 

This depends on the numerical scheme used. We currently used simple forward Euler, but some more advanced schemes may change that.

>e you can manually sub indices on the input to interpolate

This is what I was trying to avoid when added `u_t, p_t`. May sound trivial for you or me but most users won't have any idea what's going on wihtout it.

File: examples/seismic/acoustic/operators.py [hunk 1]
```diff
@@ -46,15 +46,11 @@ def ForwardOperator(model, source, receiver, time_order=2, space_order=4,
     eqn = [Eq(u.forward, stencil)]
 
     # Construct expression to inject source values
-    # Note that src and field terms have differing time indices:
-    #   src[time, ...] - always accesses the "unrolled" time index
-    #   u[ti + 1, ...] - accesses the forward stencil value
-    ti = u.indices[0]
-    src_term = src.inject(field=u, u_t=ti + 1, offset=model.nbpml,
-                          expr=src * dt**2 / m, p_t=time)
+    src_term = src.inject(field=u.forward, expr=src * dt**2 / m,
```
Comment (mloubout) [inline examples/seismic/acoustic/operators.py]: doesn't this work even without changing `pointdata`?

Comment (mlange05) [inline examples/seismic/acoustic/operators.py]: Yes, but now the feature is unused and untested, and hence assumed to be broken. I really don't get your apprehension here - it's a simple `subs` that can easily added back in by hand for special cases without having a dedicated API in core abstractions, eg. `src.inject(field=u.subs(t, ti + 2), ...)`. If those special cases become frequent, it's also easy to put it back in - but until that time it's just API clutter.

**Comment by mlange05** [comment]
> This is what I was trying to avoid when added u_t, p_t. May sound trivial for you or me but most users won't have any idea what's going on wihtout it.

Ok, I hear you. I was referring to using explicit `subs` for development purposes, if and when we play with staggered and "weird setups". If we happen to find a good reason to do these `subs` often and regularly, we can easily put the short-cut back in - but until such a time having an unused backdoor seems overkill, especially we have a still reasonably simple developer-level workaround.

Also, if you're happy with my reasoning, can you please change the review status?

**Comment by mloubout** [comment]
Well I need to make a new Julia branch and make compatible with this first because if we merge that SLIM crashes, and fined time to do it.

**Comment by mlange05** [comment]
@mloubout Any update on down-streaming this change to Julia?

Also, in order to avoid this sort of delay for interface changes, do you maybe want to make the Julia install use explicit Devito version tags? We've recently done several releases, so you can always run on a safe version and update Julia, whenever you think it needs updating and have the time to align the two.

**Comment by mloubout** [comment]
@mlange05 
SO I actually need at least `p_t` as I am working with kernel that require:
`rec.inject(v.backward, p_t=time+shift)` with shift betwee ~-200:+200

**Comment by mlange05** [comment]
Ok, fine. I'll rebase and leave the backdoors in. But please be warned that after this merge they will effectively be untested and can thus break at any point without us catching it.

**Comment by mlange05** [comment]
@mloubout Can you please verify this still works for you, so I can finally merge it? Thanks

**Comment by mloubout** [comment]
Good to go.

**Comment by mlange05** [comment]
Great, merging...

**Pull request merged**